### ISSUE TITLE
Defining a new Multi-Task Problem

### ISSUE THREAD

**Issue by agemagician** [tensorflow/tensor2tensor#1379]
Hello,

I am trying to define a new multi-task problem that predict the secondary structure protein from amino acids. I am using the transformer base model.

I am facing several problems/questions:

1) After training, when I change the decoding problem id, I get the same result for both problems. The output doesn't change. Is there a bug in the multi-task decoding ?
2) How I can force the decoder to always produce the same length as the input ?
3) With the current code I can find a lot of the input vocabulary in the output vocabulary. It extremely reduces the accuracy. How can I force the model not to share vocabulary?
4) The accuracy is really low compared to a simple CNN that I made in pytorch. Is there something wrong in the Code bellow ?

This is my code unlabelled data "like language modelling":
```
@registry.register_problem
class LanguagemodelUniref50C8k(text_problems.Text2SelfProblem):

  @property
  def approx_vocab_size(self):
    return 2**13  # 8192

  def is_generate_per_split(self):
    return False

  @property
  def dataset_splits(self):
    return [{
        "split": problem.DatasetSplit.TRAIN,
        "shards": 99,
    }, {
        "split": problem.DatasetSplit.EVAL,
        "shards": 1,
    }]

  def generate_samples(self, data_dir, tmp_dir, dataset_split):
    filepath = tmp_dir + '/uniref50_protein_unlabeled.txt'
    for line in tf.gfile.Open(filepath):
        yield {"targets": line}
```

This is my code to the labeled dataset:

```
@registry.register_problem
class TranslateAminoProtinTokensSharedVocab(text_problems.Text2TextProblem):

  @property
  def approx_vocab_size(self):
    return 2**13  # 8192

  @property
  def is_generate_per_split(self):
    return False

  @property
  def dataset_splits(self):
    return [{
        "split": problem.DatasetSplit.TRAIN,
        "shards": 9,
    }, {
        "split": problem.DatasetSplit.EVAL,
        "shards": 1,
    }]

  def generate_samples(self, data_dir, tmp_dir, dataset_split):
       datasetdf = pd.read_csv(tmp_dir + '/complete_train_dataset_seperated.csv')
       for index, row in datasetdf.iterrows():
         yield {
         "inputs": row['input'],
         "targets": row['output'],
         }
```

This is my code for the Multi-task:
```
@registry.register_problem
class MultiUniref50C8kTranslateAminoProtin(multi_problem.MultiProblem):

  def __init__(self, was_reversed=False, was_copy=False):
    super(MultiUniref50C8kTranslateAminoProtin, self).__init__(was_reversed, was_copy)
    self.task_list.append(uniref.LanguagemodelUniref50C8k())
    self.task_list.append(translate_amino_protin.TranslateAminoProtinTokensSharedVocab())

  @property
  def vocab_type(self):
    return text_problems.VocabType.SUBWORD
```

**Comment by stefan-falk** [comment]
@agemagician Hi! Have you been able to fix these issues? I am currently trying to implement a `MultiProblem` myself.

**Comment by agemagician** [comment]
The multi-task problem is still in beta version, it is still buggy and without proper in depth documentation.

I would wait a couple of months until they fix the bugs and provide a proper documentation.

Regarding the problems that I encountered because I didn't use shared vocabulary between the language modelling and the supervised task. Regarding the decoder I couldn't force the decoder to produce the input length, but rather than the maximum of length of the model output.

Feel free to open a new issue with your source Code as well as your specific problems and I will try ti help you out.

**Comment by stefan-falk** [comment]
@agemagician I see, thank you for your feedback.
